### PR TITLE
CCD-8236 globalsearch next hearing changes

### PR DESCRIPTION
**Pull request by markdathornehmcts**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/CCD-8236

### Change description ###

Add next hearing details to globalSearch

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: logstash/pipeline/02_filter.conf [hunk 1]
```diff
@@ -97,6 +104,13 @@ filter {
                 }
             }
         }
+        if [data_classification][nextHearingDetails] {
```
Comment (patelila) [inline logstash/pipeline/02_filter.conf:107]: High

  - Existing global-search documents may not contain the new field.

    The Logstash clone now preserves nextHearingDetails at ccd-docker/logstash/pipeline/02_filter.conf:107. This only helps documents
    processed after the change.

    If existing global-search documents are not reindexed or updated, MC  (Manage Case) may still receive no nextHearingDate for existing cases. This
    directly risks the issue described.

    Required action:
      - Provide a reindex/backfill plan for global_search.
      - Confirm whether the backfill happens before MC switches from Elasticsearch to /globalSearch.
      - Define the rollout order: mapping → Logstash → reindex → MC switch.
      -

Comment (patelila) [inline logstash/pipeline/02_filter.conf:107]: Low
  - No Logstash test was added.

    There is no automated verification that the cloned SearchCriteria event retains:
      - data.nextHearingDetails;
      - data_classification.nextHearingDetails;
      - both child fields, including hearingDateTime.

    Add a fixture/test for valid, missing, null, and malformed values.

There is no Logstash-specific test harness in ccd-docker. Please cover this through the existing ccd-data-store-api AAT F-1005 flow
by adding nextHearingDetails to the prerequisite case and asserting nextHearingDate in the /globalSearch response. This will verify
the complete case → Logstash → Elasticsearch → API path.

Comment (markdathornehmcts) [reply logstash/pipeline/02_filter.conf:107]: done

Comment (markdathornehmcts) [reply logstash/pipeline/02_filter.conf:107]: Covered through the existing data-store F-1005 AAT flow. The new scenarios create a case containing nextHearingDetails, wait for Logstash, and assert nextHearingDate through /globalSearch. A second scenario verifies child-level ACL filtering. Missing, null, and malformed behavior is covered by the existing transformer and Elasticsearch integration tests; malformed DateTime values cannot be created through the normal CCD case API.
